File: Problem_1.py
# Time Complexity: O(n)
# Space Complexity: O(n)
# Did this problem run on Leetcode: Yes
# Any issues faced doing this problem: No

class Solution(object):
    def exclusiveTime(self, n, logs):
        """
        :type n: int
        :type logs: List[str]
        :rtype: List[int]
        """
        tm = ([0] * n)
        st = []
        latestTime = 0
        for i in logs:
            ar = i.split(":")
            if ar[1] == "start":
                if len(st) != 0:
                    tm[int(st[-1][0])] += (int(ar[2]) - latestTime)
                st.append((ar[0], ar[2]))
                latestTime = int(ar[2])
            else:
                tm[int(ar[0])] += (int(ar[2]) - latestTime + 1)
                st.pop()
                latestTime = (int(ar[2]) + 1)
        return tm

File: Problem1.cs
public class Solution
	{
        // Time Complexity : O(n)
        // Space Complexity : O(n)
        // Did this code successfully run on Leetcode : Yes
        // Any problem you faced while coding this : No 
        public int[] ExclusiveTime(int n, IList<string> logs)
        {
            int prevTime = 0;
            Stack<int> st = new Stack<int>();
            int[] result = new int[n];
            foreach (var log in logs)
            {
                string[] splitLog = log.Split(':');
                int currTaskId = int.Parse(splitLog[0]);
                int currTime = int.Parse(splitLog[2]);

                if (splitLog[1] == "start")
                {
                    if (st.Count > 0)
                    {
                        result[st.Peek()] += currTime - prevTime;
                    }
                    prevTime = currTime;
                    st.Push(currTaskId);
                }
                else
                {
                    int popped = st.Pop();
                    result[popped] += (currTime + 1) - prevTime;
                    prevTime = currTime + 1;
                }
            }
            return result;
        }
    }
